pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -30,26 +30,45 @@ var UnAffectedConfigKinds = map[model.NodeType]sets.Set[kind.Kind]{
 	model.SidecarProxy: sets.New(kind.Gateway, kind.KubernetesGateway),
 }
 
-// ConfigAffectsProxy checks if a pushEv will affect a specified proxy. That means whether the push will be performed
-// towards the proxy.
-func ConfigAffectsProxy(req *model.PushRequest, proxy *model.Proxy) bool {
-	// Empty changes means "all" to get a backward compatibility.
+// filterRelevantUpdates filters PushRequest.ConfigsUpdated so that only configs relevant to the proxy are included,
+// and returns a new PushRequest with the filtered configs.
+func filterRelevantUpdates(proxy *model.Proxy, req *model.PushRequest) *model.PushRequest {
```

Comment:
nit: I would mention the returned PushRequest is not necessarily a copy so should not be mutated

Reply:
done